src/when.test.ts

```diff
@@ -0,0 +1,181 @@
+import { constant } from "./constant";
+import { isDefined } from "./isDefined";
+import { isNot } from "./isNot";
+import { isNullish } from "./isNullish";
+import { isStrictEqual } from "./isStrictEqual";
+import { isString } from "./isString";
+import { map } from "./map";
+import { pipe } from "./pipe";
+import { when } from "./when";
+
+describe("dataFirst", () => {
+  describe("without else", () => {
+    it("returns the happy path when true", () => {
+      expect(when("hello", isStrictEqual("hello"), constant("was true"))).toBe(
+        "was true",
+      );
+    });
+
+    it("returns the identity when false", () => {
+      expect(when("hello", isStrictEqual("olleh"), constant("was true"))).toBe(
+        "hello",
+      );
+    });
+
+    it("passes extra args to the functions", () => {
+      expect(
+        when(
+          "123",
+          isString,
+          (x, radix) => Number.parseInt(x, radix),
+          // Extra args:
+          10 /* radix */,
+        ),
+      ).toBe(123);
```

Comment:
**suggestion:** this wouldn't catch if the `10` wasn't being passed

```suggestion
          4 /* radix */,
        ),
      ).toBe(27);
```

Reply:
Good catch